CONTRIBUTING.md

```diff
@@ -0,0 +1,132 @@
+# Contributing to SambaNova AI Starter Kits
+
+Thank you for your interest in contributing to SambaNova AI Starter Kits! This document provides guidelines and best practices for contributing to our repository.
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Getting Started](#getting-started)
+- [Branch Protection Rules](#branch-protection-rules)
+- [Development Workflow](#development-workflow)
+- [Code Quality Standards](#code-quality-standards)
+- [Pull Request Guidelines](#pull-request-guidelines)
+- [Package Owner Responsibilities](#package-owner-responsibilities)
+
+## Overview
+
+SambaNova AI Starter Kits are open-source examples and guides designed to facilitate the deployment of AI-driven use cases. We welcome contributions that improve existing kits or add new functionality.
+
+## Getting Started
+
+1. Create a branch (not a fork) for your contribution
+2. Set up your development environment
+3. Make your changes following our coding standards
+4. Submit a pull request
+
+## Branch Protection Rules
+
+The `main` branch is protected with the following rules:
+
+- Pull requests must be up-to-date with the base branch
+- All conversations must be resolved before merging
+- At least one reviewer approval is required
+- All GitHub Actions checks must pass
+
+## Development Workflow
+
+### Branch Naming Conventions
+
+Use the following prefixes for your branches:
+
+- `feature/` for new features (e.g., `feature/rag-component`)
+- `improvement/` for improvements (e.g., `improvement/add-langchain-cache`)
+- `bugfix/` for bug fixes (e.g., `bugfix/vectorstore`)
+- `documentation/` for documentation (e.g., `documentation/rag-component`)
+- `release/` for releases (e.g., `release/v1.0.1`)
+
+### Code Quality Standards
+
+#### Format and Lint Requirements
+
+All code must pass our automated checks using Ruff and MyPy. You can run these checks locally using:
+
+```bash
+# Using Makefile (Preferred)
+make venv-install  # Initialize environment
+make format       # Format code with Ruff
+make lint         # Lint and type check with Ruff & MyPy
+make format-lint  # Run all checks
+
+# Direct commands (Optional)
+ruff format your_module
+ruff check --fix your_module
+ruff check --fix --select I your_module
+mypy --explicit-package-bases your_module
+```
+
+#### Testing Requirements
+
+- All contributions must pass the unit test suite
```

Comment:
New aisk modules (at high effort level with UI, etc.) should have some test coverage that has been reviewed.